redux/reducersComp/cryptosSlice.ts

```diff
@@ -0,0 +1,85 @@
+import {createSlice, createAsyncThunk} from '@reduxjs/toolkit';
+import type {PayloadAction} from '@reduxjs/toolkit';
+import axios from 'axios';
+
+import type {RootState} from '../store';
+import {initialStateProps, DataCriptoInfo} from '../../utils/interfaces';
+
+const initialState: initialStateProps = {
+  cryptosData: [],
+  error: '',
+};
+
+export const cryptoApiData = createAsyncThunk(
+  'crypto/api',
+  async (name: string) => {
+    const response = await axios.get(
+      `https://data.messari.io/api/v1/assets/${name}/metrics`,
```

Comment:
keep this: https://data.messari.io/api/v1/
as the base url, stored in a .env file. 
https://yarnpkg.com/package/react-native-dotenv